snippets/component-properties/args/octane-child.js

```diff
@@ -0,0 +1,6 @@
+// child-component.js
+import Component from '@ember/component';
+
+export default Component.extend({
+  name: 'Child Name'
+});
```

Comment:
Plus native class syntax

Reply:
If the goal is to show case the `this.args.*` syntax, it should be done in the JS class (as a getter or something).

Maybe something like this (feel free to word smith it):

```
export default class ChildComponent extends Component {
  name = 'Child Name';

  get normalizedName() {
    // In Octane, `this.args.name` refers to the passed in `@name` argument from the invocation,
    // whereas `this.name` refers to the class field defined above.
    return this.args.name || this.name;
  }
}
```

Reply:
I think Godfrey's example improves illustration a bit. However, I think we should recommend writing clear code and avoid using the same variable name for the property and argument. (The cheatsheet covers [setting a default value for an argument](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/#actions__template-arguments-default).)

Reply:
What I was trying to show here was that in the old world, `name` was overwritten. In the Octane, `name` is not overwritten in the child, it is placed in `args`. Not sure what the best way to show that in code is. Maybe we just don't show that, and we leave it in the description.